src/app/project/ProjectDetails.js

```diff
@@ -35,6 +39,23 @@ define([
 
             this.statusClass = params.status.replace(' ', '');
         },
+        postCreate: function () {
+            // summary:
+            //      description
+            // param or return
+            console.log('app.project.ProjectDetails:postCreate', arguments);
+
+            if (this.allowEdits) {
+                var btn = domConstruct.create('button', {
+                    role: 'button',
+                    'class': 'btn btn-primary',
+                    innerHTML: 'Add Feature'
+                }, this.toggleNode.parentNode, 'first');
+
+                var that = this;
```

Comment:
This has no effect on the context that `onAddFeatureClick` is called with. You'll noticed that `this` is bound to the button when `onAddFeatureClick` is called. This would happen if you passed `that.onAddFeatureClick` or `this.onAddFeatureClick` since you are not invoking the method when you pass it to `on()`. It's working currently only because `onAddFeatureClick` doesn't reference `this`. However, I still think that it should be cleaned up.

I would recommend using [bind](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) or `lang.hitch()` in this instance just to make thing more clear.